### PR TITLE
graylog support

### PR DESCRIPTION
**Pull request by drzraf**
Only works if `graylog2/gelf-php` is installed (see #1)

### CODE REVIEW

**Comment by stephan-strate** [comment]
I will give it a test in the next days. It will take some time, because I need to setup a graylog instance.
Also, you will need to add `graylog2/gelf-php` as dependency as I stated in https://github.com/stephan-strate/grav-plugin-logger-channels/pull/1#issuecomment-877237766.

**Comment by stephan-strate** [comment]
I know it feels weird, but you actually need to add the vendor files to make this work. Grav installs the source itself.

Here are some examples:

- https://github.com/getgrav/grav-plugin-admin
- https://github.com/getgrav/grav-plugin-email

**Comment by drzraf** [comment]
ping?

**Comment by stephan-strate** [comment]
Sorry for letting you wait so long. This pull request got lost in my pile of work.
Could you provide me some instructions on how to test this properly? I am not familiar with graylog and therefore have no access to a running instance.

**Comment by drzraf** [comment]
I use OVH hosted one. It's really cheap and work out of the box.

File: classes/Handlers/GraylogHandler.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        // Setup formatter
+        $formatter = new GelfMessagePrefixer();
+        $extraFields = [];
+        array_walk($options['extra_fields'], function ($item) use (&$extraFields) {
+            $extraFields[$item['key']] = $item['value'];
+        });
+        $formatter->setExtraFields($extraFields);
+
+        // Setup handler
+        $this->handler = new GelfHandler($publisher);
```
Comment (stephan-strate) [inline classes/Handlers/GraylogHandler.php]: We need the log level to be configurable. Have a look at eg. the TelegramHandler.

**Review by stephan-strate — CHANGES_REQUESTED** [review]
Looks good to me so far. One small change and this is done :)

**Pull request merged**